core/good_key.go

```diff
@@ -42,36 +42,41 @@ var (
 // key use (our requirements are the same for either one), according to basic
 // strength and algorithm checking.
 // TODO: Support JsonWebKeys once go-jose migration is done.
-func GoodKey(key crypto.PublicKey) bool {
+func GoodKey(key crypto.PublicKey, maxKeySize int) bool {
 	log := blog.GetAuditLogger()
 	switch t := key.(type) {
 	case rsa.PublicKey:
-		return GoodKeyRSA(t)
+		return GoodKeyRSA(t, maxKeySize)
 	case *rsa.PublicKey:
-		return GoodKeyRSA(*t)
+		return GoodKeyRSA(*t, maxKeySize)
 	case ecdsa.PublicKey:
-		return GoodKeyECDSA(t)
+		return GoodKeyECDSA(t, maxKeySize)
 	case *ecdsa.PublicKey:
-		return GoodKeyECDSA(*t)
+		return GoodKeyECDSA(*t, maxKeySize)
 	default:
 		log.Debug(fmt.Sprintf("Unknown key type %s", reflect.TypeOf(key)))
 		return false
 	}
 }
 
-func GoodKeyECDSA(key ecdsa.PublicKey) bool {
+func GoodKeyECDSA(key ecdsa.PublicKey, maxKeySize int) bool {
 	log := blog.GetAuditLogger()
 	log.Debug(fmt.Sprintf("ECDSA keys not yet supported."))
 	return false
 }
 
-func GoodKeyRSA(key rsa.PublicKey) bool {
+func GoodKeyRSA(key rsa.PublicKey, maxKeySize int) bool {
```

Comment:
In a future patch, we should make these return error instead of bool, so we can tell people that you can't use a 16384-bit RSA key.